jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -233,6 +234,13 @@ class S3TreeTable(
         }
     }
 
+    override fun getData(dataId: String): Any? = when {
+        S3EditorDataKeys.SELECTED_NODES.`is`(dataId) -> this.getSelectedNodes().filterNot { it is S3TreeContinuationNode<*> }
+        S3EditorDataKeys.BUCKET.`is`(dataId) -> this.bucket
+        S3EditorDataKeys.BUCKET_VIEWER.`is`(dataId) -> this
+        else -> null
```

Comment:
throw instead of null?

Reply:
Nope, API contract says return null

Reply:
Ah, sure. It's kind of weird though because that is something we wouldn't want to silently fail if we messed up

Reply:
DataProvider traverses up the UI tree, so null means ask the parent.